stdlib/src/builtin/simd.mojo

```diff
@@ -2192,6 +2192,24 @@ struct SIMD[type: DType, size: Int = simdwidthof[type]()](
             "llvm.vector.reduce.or", Scalar[DType.bool], has_side_effect=False
         ](self)
 
+    @always_inline
+    fn reduce_all(self) -> Bool:
```

Comment:
Do we really want to make them public? Also, maybe `@paramter if` on `DType.bool` to prevent generating redundant IR?

Reply:
+1 on both points. I'd lazily make them public if desired in the future. Also agree re parameter if.

Reply:
Actually, I guess we should probably add the parameter if to `cast`, i.e. making it a no-op when `target_type == type`.

Reply:
I did it this way to match the other reduce functions, but since it's better to use the builtin `any`/`all` functions in this case, making them private might make sense.

Reply:
thanks for the suggestions